scanner.cs

```diff
@@ -3,66 +3,176 @@
+    public void initializeScanner(string fileName) {
+        try {
+            if(!fileName.EndsWith(".mp")) {
+                // File format exception
+                throw new Exception(Constants.ERROR_FILE_FORMAT);
+            } else {
+                // Grab all bytes from the file
+                StreamReader reader = new StreamReader(fileName);
+                string temp = reader.ReadToEnd();
+                __bytes = temp.ToCharArray();
 
-				// Caching variables to save memory
-				int i = 0,
-					length = __bytes.Length;
-				string ws = Constants.WHITESPACE;
+                // Caching variables to save memory
+                int i = 0,
+                    length = __bytes.Length;
+                string ws = Constants.WHITESPACE;
 
-				// Loop through, ignoring whitespace
-				for(i = 0; i < length; i++) {
-					if(ws.Contains("" + __bytes[i])) {
-						continue;
-					} else {
-						getNextToken();
-					}
-				}
-			}
-		} catch(Exception ex) {
-			Console.WriteLine(ex);
-		}
-	}
+                // Loop through, ignoring whitespace
+                for(i = 0; i < length; i++) {
+                    if(ws.Contains("" + __bytes[i])) {
+                        if(__bytes[i] == '\n'){
+                            line++;
+                        }
+                        continue;
+                    } else {
+                        getNextToken();
+                    }
+                }
+            }
+        } catch(Exception ex) {
+            Console.WriteLine(ex);
+        }
+    }
 
-	// This is the dispatcher
-	private int getNextToken() {
-		return 0; // change this
-	}
+    // This is the dispatcher
+    private int getNextToken() {
+        return 0; // change this
+    }
 
-	private int getLineNumber() {
-		return 0; // change this
-	}
+    private int getLineNumber() {
+        return 0; // change this
+    }
 
-	private int getColumnNumber() {
-		return 0; // change this
-	}
+    private int getColumnNumber() {
+        return 0; // change this
+    }
 
-	// Finite State Automatons
-	private int fsaLetter() {
-		return 0; // change this
-	}
+    // Finite State Automatons
+    private int fsaLetter() {
+        return 0; // change this
+    }
 
-	private int fsaDigit() {
-		return 0; // change this
-	}
+    private Tuple<string, TOKENS> fsaDigit() {
+        string lexeme = "";
+        string DIGITS = Constants.DIGITS;
+        TOKENS token;
+        char next;
+        goto S0; // Smothers warning about S0 label not used
+        S0: //Start state
+            next = __bytes[curByte];
+            column++;
+            curByte++;
+            if(DIGITS.Contains("" + next)){
+                lexeme += next;
+                goto S1;
+            } else {
+                curByte--;
+                throw new Exception(
+                    "Dispatcher done goofed. Passed " +
+                    next + " to the digit FSA, but " +  next +
+                    "is not a digit."
```

Comment:
Make this error a constant string in our constants class; would need to reformat the message to be able to make the work.